src/test/util/mining.cpp

```diff
@@ -92,6 +92,11 @@ COutPoint MineBlock(const NodeContext& node, std::shared_ptr<CBlock>& block)
         assert(block->nNonce);
     }
 
+    return ConnectBlock(node, block);
+}
+
+COutPoint ConnectBlock(const NodeContext& node, const std::shared_ptr<CBlock>& block)
```

Comment:
I'd prefer to call this wrapper around `ProcessNewBlock` some other name (e.g. `ProcessBlock()` like `net_processing` does) to avoid confusion with the existing `ConnectBlock` in validation. The added function doesn't necessarily attempt to connect the block, for example if a conflicting block arrived / was connected in between mining and calling `ProcessNewBlock` the real `ConnectBlock` may never be called.

Reply:
Done.